client/allocrunner/taskrunner/stats_hook.go

```diff
@@ -99,7 +99,15 @@ MAIN:
 		case ru, ok := <-ch:
 			// if channel closes, re-establish a new one
 			if !ok {
-				goto MAIN
+				// backoff if driver closes channel, potentially
+				// because task shutdown or because driver
+				// doesn't implement channel interval checking
+				select {
+				case <-time.After(h.interval):
+					goto MAIN
+				case <-ctx.Done():
+					return
+				}
```

Comment:
I've decided to unilaterily back off with interval regardless of the reason.  We could be a bit more discerning and wait for a fraction of interval, or only what's left of the interval, but I didn't see the additional complexity worth it, as this is a workaround a driver misbehavior case.